src/components/TodoItem.tsx

```diff
@@ -0,0 +1,192 @@
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const myInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    myInput.current && myInput.current.focus();
+  }, [isEditing, myInput]);
+
+  const handleEditStart = () => {
+    setIsEditing(true);
+  };
+
+  const handleEditCancel = () => {
+    setIsEditing(false);
+    setNewTitle(title);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodo(todo);
+
+    deleteTodo(todoId)
+      .then(() => setTodos((prev) => prev.filter((t) => t.id !== todoId)))
+      .catch(() => setErrorMessage(Error.Delete))
+      .finally(() => setLoadingTodo(null));
+  };
+
+  const handleChangeCheckbox = (todoId: number) => {
+    setLoadingTodo(todo);
+
+    updateTodo(todoId, { completed: !completed })
+      .then(() => setTodos(prev => prev
+        .map(newTodo => (newTodo.id === todoId
+          ? ({ ...newTodo, completed: !completed })
+          : newTodo))))
+      .catch(() => setErrorMessage(Error.Update))
+      .finally(() => {
+        setLoadingTodo(null);
+      });
+  };
+
+  const handleEditSubmit = (event: EditEvent = null) => {
+    const newTrimmedTitle = newTitle?.trim();
+
+    if (event) {
+      event.preventDefault();
+    }
+
+    if (!newTrimmedTitle) {
+      handleDeleteTodo(todo.id);
+
+      return;
+    }
+
+    if (newTitle === title) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    setLoadingTodo(todo);
+
+    updateTodo(todo.id, { title: newTrimmedTitle })
+      .then((newTodo: Todo) => {
+        setTodos((prev) => prev.map((t) => (t.id === todo.id ? newTodo : t)));
+        setIsEditing(false);
+      })
+      .catch(() => {
+        setErrorMessage(Error.Update);
+      })
+      .finally(() => {
+        setLoadingTodo(null);
+      });
+  };
+
+  const handleKeyUp: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+    if (event.key === 'Enter') {
+      handleEditSubmit();
+    } else if (event.key === 'Escape') {
+      handleEditCancel();
+    }
+  };
+
+  return (
+    isEditing ? (
+      <div data-cy="Todo" className="todo">
+        <label className="todo__status-label">
+          <input
+            data-cy="TodoStatus"
+            type="checkbox"
+            className="todo__status"
+          />
+        </label>
+
+        <form onSubmit={(e) => e.preventDefault()}>
+          <input
+            onKeyUp={handleKeyUp}
+            ref={myInput}
+            onBlur={() => handleEditSubmit()}
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={newTitle}
+            onChange={(event) => setNewTitle(event.target.value)}
+          />
+        </form>
+
+        <div
+          data-cy="TodoLoader"
+          className={classNames(
+            'modal overlay',
+            { 'is-active': loadingTodo === todo },
```

Comment:
currently the loaders doesn't appear when you have multiple changes

for example when you toggle all todos (check the working version)

it should be like that
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/9ad6ab12-1f62-4d98-a6e5-4b133aba5393)

in order to do that you need to save in `loadingTodo` not only one todo, but have an array of todos and add 'is-active' for each todo from this array